src/core/connection.h

```diff
@@ -563,6 +564,18 @@ typedef struct QUIC_CONNECTION {
     //
     QUIC_LOSS_DETECTION LossDetection;
 
+    //
+    // ECT and CE counters for different packet number spaces.
+    //
+    uint64_t EcnEctCounters[QUIC_ENCRYPT_LEVEL_COUNT];
+    uint64_t EcnCeCounters[QUIC_ENCRYPT_LEVEL_COUNT]; // maps to ecn_ce_counters in RFC 9002.
+
+    //
+    // The total number of packets sent with each corresponding ECT codepoint in all encryption
+    // level.
+    //
+    uint64_t NumPacketsSentWithEct;
```

Comment:
Can this be moved to a sub struct? I feel like it shouldn't be on the connection itself.

Reply:
Should it be per-path?

Reply:
The ECN count feedback isn't per path, right? If we make this per-path, it will be unsync'd with ECN counts from the peer.